### PR TITLE
Add onstage offstage support

### PR DESCRIPTION
**Pull request by rehlma**
By default, `spot()` only finds widgets that are "onstage", not hidden with the [`Offstage`](https://api.flutter.dev/flutter/widgets/Offstage-class.html) widget.

To find offstage widgets, start your widget selector with `spotOffstage()`.
Search for both - the on- and offstage widgets - with `spotAllWidgets()`.

For existing selectors, use `overrideWidgetPresence(WidgetPresence presence)` to modify the presence to `offstage`, `onstage` or `combined`.

```dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:spot/spot.dart';

void main() {
  testWidgets('Spot offstage and combined widgets', (tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: Row(
          children: [
            Text('a'),
            Text('b'),
            Offstage(child: Text('c')),
          ],
        ),
      ),
    );
    
    spot<Text>().withText('a').existsOnce();
    spot<Text>().withText('c').doesNotExist();
    spot<Text>().withText('c').overrideWidgetPresence(WidgetPresence.offstage).existsOnce();
    
    spotOffstage().spot<Text>().atMost(3);
    spotOffstage().spotText('c').existsOnce();
    spotOffstage().overrideWidgetPresence(WidgetPresence.onstage).spotText('c').doesNotExist();
    
    spotAllWidgets().spotText('a').existsOnce();
    spotAllWidgets().spotText('c').existsOnce();
    spotOffstage().overrideWidgetPresence(WidgetPresence.combined).spotText('a').existsOnce();
    spotOffstage().overrideWidgetPresence(WidgetPresence.combined).spotText('c').existsOnce();
  });
}
```

### CODE REVIEW

File: lib/src/spot/tree_snapshot.dart [hunk 1]
```diff
@@ -23,19 +23,44 @@ WidgetTreeSnapshot createWidgetTreeSnapshot() {
...
+        isParentOffstage: isOffstage,
+      );
+      node.addChild(childNode);
     }
-    return snapshot;
+    return node;
   }
 
-  final origin = build(rootElement);
+  // TODO replace recursion with iteration to prevent stack overflow for giant widget trees
```
Comment (passsy) [inline lib/src/spot/tree_snapshot.dart:62]: @rehlma Haven't you already done this?

File: lib/src/spot/selectors.dart [hunk 2]
```diff
@@ -49,19 +64,61 @@ mixin ChainableSelectors<T extends Widget> {
...
+  ///
+  /// Returns a new [WidgetSelector] that includes offstage widgets.
+  ///
+  /// ### Example usage:
+  /// ```dart
+  /// final text = spotText('text')
+  ///   .overrideIncludeOffstage();
+  /// ```
+  @useResult
+  WidgetSelector<T> overrideIncludeOffstage(bool offstage) {
```
Comment (passsy) [inline lib/src/spot/selectors.dart:90]: @rehlma should we use an `enum { onstage, offstage, both }`?

Comment (rehlma) [reply lib/src/spot/selectors.dart:90]: What about `enum VisibilityStage { onstage, offstage, all }`?

Comment (passsy) [reply lib/src/spot/selectors.dart:90]: `enum WidgetPresence  { onstage, offstage, combined }` 👌

**Pull request merged**